src/p09.rs

```diff
@@ -0,0 +1,87 @@
+use std::{
+    collections::{HashSet, VecDeque},
+    error::Error,
+};
+
+fn get_neighbor_points(map: &[String], pos: (usize, usize)) -> Vec<(usize, usize)> {
+    let mut neighbors = Vec::new();
+
+    if pos.0 > 0 {
+        neighbors.push((pos.0 - 1, pos.1));
+    }
+    if pos.0 < map.len() - 1 {
+        neighbors.push((pos.0 + 1, pos.1));
+    }
+    if pos.1 > 0 {
+        neighbors.push((pos.0, pos.1 - 1));
+    }
+    if pos.1 < map[0].len() - 1 {
+        neighbors.push((pos.0, pos.1 + 1));
+    }
+
+    neighbors
+}
+
+fn get_low_points(map: &[String]) -> Vec<(usize, usize)> {
+    let mut low_points = Vec::new();
+
+    for y in 0..map.len() {
+        for x in 0..map[0].len() {
+            if get_neighbor_points(map, (y, x))
+                .iter()
+                .all(|p| map[p.0].chars().nth(p.1).unwrap() > map[y].chars().nth(x).unwrap())
+            {
+                low_points.push((y, x));
+            }
+        }
+    }
+
+    low_points
+}
+
+fn get_basin_size(map: &[String], start: (usize, usize)) -> i64 {
+    let mut visited = HashSet::new();
+    let mut queue = VecDeque::new();
+    queue.push_back(start);
+
+    while !queue.is_empty() {
+        let pos = queue.pop_front().unwrap();
+        if !visited.contains(&pos) {
+            for next in get_neighbor_points(map, pos).iter().filter(|(y, x)| {
+                map[*y].chars().nth(*x).unwrap() != '9' && !visited.contains(&(*y, *x))
+            }) {
```

Comment:
```suggestion
            for next in get_neighbor_points(map, pos).into_iter().filter(|(y, x)| {
                map[y].chars().nth(x).unwrap() != '9' && !visited.contains(&(y, x))
            }) {
```
maybe

Reply:
I doesn't really work like that:
```
error[E0277]: the type `[String]` cannot be indexed by `&usize`
  --> src/p09.rs:51:17
   |
51 |                 map[y].chars().nth(*x).unwrap() != '9' && !visited.contains(&(*y, *x))
   |                 ^^^^^^ slice indices are of type `usize` or ranges of `usize`
   |
   = help: the trait `SliceIndex<[String]>` is not implemented for `&usize`
   = note: required because of the requirements on the impl of `Index<&usize>` for `[String]`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `aoc2021` due to previous error
```

Putting `into_iter()` there only allows me to use `queue.push_back(next);` instead of `queue.push_back(*next);`.